Fix(parser): Fix the bug causing tests to fail regarding matrices in the parser, modify config files to use correct native parser tag, change parser to find child nodes by field name rather than index where possible

leiamcalisteryoung opened the pull request:
There are a lot of tests failing but it isn't at the parsing stage, it is at stage 2a (rewritten model). I am not sure what I changed that is causing these tests to fail.

leiamcalisteryoung commented:
On my computer the tests are failing during the integration tests but on here it seems the tests related to traversal.rs in the parser crate are failing. Do you know anything about this or what might be causing it? @gskorokhod

gskorokhod commented:
> On my computer the tests are failing during the integration tests but on here it seems the tests related to traversal.rs in the parser crate are failing. Do you know anything about this or what might be causing it? @gskorokhod 

Hey! Not really. I'll have a look ASAP - some of the stuff in traversal might not even be necessary anymore tbh

TAswan commented:
I suspect the issue is from your most recent merge with main.

Using the mod 6 test case as an example (on conjure's parser, not yours), both commit **963d1053 change another use_native_parser** (before merge) as well as the current main agree on the solution with conjure (see human rule trace in good.txt) whereas commit **65044e1c formatting** (the merge itself doesn't run) has issues with expanding the negated_eq_to_neq  rule at least (see bad.txt). 

Using git diff, the main suspects seem to be
 crates/conjure_essence_parser/src/parser/{
domain.rs,
 expression.rs,
find.rs,
 letting.rs,
 parse_model.rs} 

[good.txt](https://github.com/user-attachments/files/19833462/good.txt)
[bad.txt](https://github.com/user-attachments/files/19833464/bad.txt)

leiamcalisteryoung commented:
I reverted the change in build.rs. I don't think it does anything meaningful but I reverted it just in case. The other changes in that commit are all within the parser crate which shouldn't be used at all during testing since the native parser isn't enabled.

leiamcalisteryoung commented:
With the traversal tests commented out, the tests are now showing what I see when I run cargo test which is that many of the integration tests are failing either at stage 2a or while building the Minion model.

leiamcalisteryoung commented:
PR
